plugins/modules/zendesk_ticket.py

```diff
@@ -0,0 +1,294 @@
+            subject (str): The subject of the ticket.
+
+        Returns:
+            dict: A dictionary containing the result of the ticket creation operation.
+        """
+        changed = False
+        url = '{0}/api/v2/tickets'.format(self.host)
+        payload = {
+            "ticket": {
+                "comment": {
+                    "body": body
+                },
+                "priority": priority,
+                "subject": subject
+            }
+        }
+
+        request = self.api_auth()
+
+        try:
+            response = request.post(url, data=json.dumps(payload))
+            if response.getcode() == 201:
+                changed = True
+                return {
+                    'changed': changed,
+                    'response': json.load(response),
+                }
+        except Exception as e:
+            return {
+                'changed': changed,
+                'msg': to_native(e)
+            }
+
+    def close_ticket(self, ticket_id, status, body):
+        """
+        Closes or resolves a ticket in Zendesk.
+
+        Args:
+            ticket_id (int): The ID of the ticket to be closed or resolved.
+            status (str): The new status for the ticket ('closed' or 'resolved').
+            body (str): An optional comment to add to the ticket.
+
+        Returns:
+            dict: A dictionary containing the result of the ticket update operation.
+        """
+        url = '{0}/api/v2/tickets/{1}'.format(self.host, ticket_id)
+        payload = {
+            "ticket": {
+                "status": status,
+                "comment": {
+                    "body": body
+                }
+            }
+        }
+
+        request = self.api_auth()
+
+        try:
+            response = request.patch(url, data=json.dumps(payload))
+            if response.getcode() in [200, 204]:
+                return {
+                    'changed': True,
+                    'response': json.load(response),
+                }
+        except Exception as e:
+            return {
+                'changed': False,
+                'msg': to_native(e)
+            }
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(type='str', required=True),
+            username=dict(type='str', required=True, aliases=['user']),
+            password=dict(type='str', required=False, aliases=['pass'], no_log=True),
+            status=dict(type='str', required=True, choices=['new', 'closed', 'resolved']),
+            body=dict(type='str', default=''),
+            priority=dict(type='str', choices=['urgent', 'high', 'normal', 'low'], default='normal'),
+            subject=dict(type='str'),
+            token=dict(type='str', required=False, no_log=True),
+            ticket_id=dict(type='int')
+        ),
+        required_if=[
+            ('status', 'new', ('priority', 'subject')),
```

Comment:
`priority` is redundant here, because it has a default value - it will **always** be set.

Reply:
That was not solved.